sagemcom_api/action_error_exception_handler.py

```diff
@@ -0,0 +1,75 @@
+"""Logic to spot and create ActionErrorExceptions"""
+
+from sagemcom_api.const import XMO_ACCESS_RESTRICTION_ERR, XMO_AUTHENTICATION_ERR, XMO_LOGIN_RETRY_ERR, XMO_MAX_SESSION_COUNT_ERR, XMO_NO_ERR, XMO_NON_WRITABLE_PARAMETER_ERR, XMO_REQUEST_ACTION_ERR, XMO_UNKNOWN_PATH_ERR
+from sagemcom_api.exceptions import AccessRestrictionException, AuthenticationException, LoginRetryErrorException, MaximumSessionCountException, NonWritableParameterException, UnknownException, UnknownPathException
+
+
+class ActionErrorHandler:
+    """Raised when a requested action has an error"""
+
+    KNOWN_EXCEPTIONS = (
+        XMO_AUTHENTICATION_ERR,
+        XMO_ACCESS_RESTRICTION_ERR,
+        XMO_NON_WRITABLE_PARAMETER_ERR,
+        XMO_UNKNOWN_PATH_ERR,
+        XMO_MAX_SESSION_COUNT_ERR,
+        XMO_LOGIN_RETRY_ERR
+    )
+
+    @staticmethod
+    def throw_if(response):
+        """For anywhere that needs the old single-exception behaviour"""
+
+        if response["reply"]["error"]["description"] != XMO_REQUEST_ACTION_ERR:
+            return
+
+        actions = response["reply"]["actions"]
+        for action in actions:
+
+            action_error = action["error"]
+            action_error_desc = action_error["description"]
+
+            if action_error_desc == XMO_NO_ERR:
+                continue
+
+            raise ActionErrorHandler.FromErrorDescription(action_error, action_error_desc)
+
+    @staticmethod
+    def is_unknown_exception(desc):
+        """
+        True/false if the ActionError is one of our known types
+        """
+
+        return False if desc == XMO_NO_ERR else desc not in ActionErrorHandler.KNOWN_EXCEPTIONS
+
+    @staticmethod
+    def from_error_description(action_error, action_error_desc):
+        """
+        Create the correct exception from an error, for the caller to throw
+        """
+
+        ex = None
+
+        if action_error_desc == XMO_AUTHENTICATION_ERR:
+            ex = AuthenticationException(action_error)
```

Comment:
```suggestion
            return AuthenticationException(action_error)
```

My preference would be to not set the `ex` variable and just return it directly when raised. For all if statements here.

Reply:
Mine too ... I thought I saw "too many return statements" in the linter output and just complied.  But I think my setup was skewy for linting so I've changed it back to see if it still complains

Reply:
yeah its not happy about R0911 (too-many-returns) now, not sure which way you want to go?

Reply:
We can add an ignore for now, or change the linting preferences.

Reply:
now disabled by comment

Reply:
@iMicknl - any chance you can kick off the workflow again?  sorry about the chain of  back-and-forth